code/modules/power/lighting.dm

```diff
@@ -340,6 +340,32 @@
 	var/area/A = get_area(src)
 	return A.lightswitch && A.power_light
 
+///Flickers the lights for a set amount of time and flicker delays.
+/obj/machinery/light/proc/set_flicker(amount, flicker_time_lower, flicker_time_upper)
+	if(status == LIGHT_BROKEN)
+		return
+	///Timer id for flickering.
+	var/theflicker
+	if(flickering) //is it flickering when set_flicker-ed
+		//get rid of the stuff to start over so we dont mess stuff up.
+		deltimer(theflicker)
+		resetflickers()
+	if(!flickering) //if it is not flickering already, do as usual.
+		random_flicker = TRUE
+		flicker(TRUE)
+		if(flicker_time_lower)
+			flicker_time_lower_min = flicker_time_lower
+		if(flicker_time_upper)
+			flicker_time_upper_max = flicker_time_upper
+		theflicker = addtimer(CALLBACK(src, PROC_REF(resetflickers), amount))
```

Comment:
the timer var is used to delete the timer when its resetting lights time, it doesnt really work wrong in game not sure what you mean.

Reply:
i tried to make it better now

Reply:
Bro this is still doing nothing.
You are defining a null var at 348, then you are calling deltimer on it at 351, but theflicker var is never set to anything, so that line cannot possibly ever do anything.